lane.py

```diff
@@ -997,9 +997,12 @@ def modify_database_query(self, _db, qu):
         return qu.offset(self.offset).limit(self.size)
 
     def modify_search_query(self, search):
-        # Do nothing -- all necessary pagination information is kept in
-        # offset and size, which external_search knows how to apply.
-        return search
```

Comment:
external_search applying this information is what caused the bug. We need to handle pagination _before_ the query goes to the Elasticsearch server, not afterwards.